functests-performance-profile-creator/1_performance-profile_creator/ppc.go

```diff
@@ -0,0 +1,136 @@
+
+		for expectedProfilePath, args := range expectedProfiles {
+			cmdArgs := []string{
+				fmt.Sprintf("--disable-ht=%v", args.DisableHT),
+				fmt.Sprintf("--mcp-name=%s", args.MCPName),
+				fmt.Sprintf("--must-gather-dir-path=%s", args.MustGatherDirPath),
+				fmt.Sprintf("--power-consumption-mode=%s", args.PowerConsumptionMode),
+				fmt.Sprintf("--profile-name=%s", args.ProfileName),
+				fmt.Sprintf("--reserved-cpu-count=%d", args.ReservedCPUCount),
+				fmt.Sprintf("--rt-kernel=%v", args.RTKernel),
+				fmt.Sprintf("--split-reserved-cpus-across-numa=%v", args.SplitReservedCPUsAcrossNUMA),
+				fmt.Sprintf("--topology-manager-policy=%s", args.TMPolicy),
+				fmt.Sprintf("--user-level-networking=%v", args.UserLevelNetworking),
+			}
+
+			out, err := testutils.ExecAndLogCommand(ppcPath, cmdArgs...)
+			Expect(err).To(BeNil(), "failed to run ppc for '%s': %v", expectedProfilePath, err)
+
+			profile := &performancev2.PerformanceProfile{}
+			err = yaml.Unmarshal(out, profile)
+			Expect(err).To(BeNil(), "failed to unmarshal the output yaml for '%s': %v", expectedProfilePath, err)
+
+			bytes, err := ioutil.ReadFile(expectedProfilePath)
+			Expect(err).To(BeNil(), "failed to read the expected yaml for '%s': %v", expectedProfilePath, err)
+
+			expectedProfile := &performancev2.PerformanceProfile{}
+			err = yaml.Unmarshal(bytes, expectedProfile)
+			Expect(err).To(BeNil(), "failed to unmarshal the expected yaml for '%s': %v", expectedProfilePath, err)
+
+			Expect(profile).To(BeEquivalentTo(expectedProfile), "regression test failed for '%s' case", expectedProfilePath)
+		}
+	})
+})
+
+func getMustGatherDirs(mustGatherPath string) map[string]string {
+	Expect(mustGatherPath).To(BeADirectory())
+
+	mustGatherDirs := make(map[string]string)
+	mustGatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+	Expect(err).To(BeNil(), fmt.Errorf("can't list '%s' files: %v", mustGatherPath, err))
+
+	for _, file := range mustGatherPathContent {
+		fullFilePath := filepath.Join(mustGatherPath, file.Name())
+		Expect(fullFilePath).To(BeADirectory())
+
+		mustGatherDirs[file.Name()] = fullFilePath
+	}
+
+	return mustGatherDirs
+}
+
+func getExpectedProfiles(expectedProfilesPath string, mustGatherDirs map[string]string) map[string]cmd.ProfileCreatorArgs {
+	Expect(expectedProfilesPath).To(BeADirectory())
+
+	expectedProfilesPathContent, err := ioutil.ReadDir(expectedProfilesPath)
+	Expect(err).To(BeNil(), fmt.Errorf("can't list '%s' files: %v", expectedProfilesPath, err))
+
+	// read ppc params files
+	ppcParams := make(map[string]cmd.ProfileCreatorArgs)
+	for _, file := range expectedProfilesPathContent {
+		if filepath.Ext(file.Name()) != ".json" {
+			continue
+		}
+
+		fullFilePath := filepath.Join(expectedProfilesPath, file.Name())
+		bytes, err := ioutil.ReadFile(fullFilePath)
+		Expect(err).To(BeNil(), "failed to read the ppc params file for '%s': %v", fullFilePath, err)
+
+		ppcArgs := cmd.ProfileCreatorArgs{
```

Comment:
Let's add a comment here to explain that we are populating non-required creator args with default values. Might be worth explaining that other non-required fields that are not here: disableHT, userLevelNetworking and reservedCPUCount automatically default to false, false and 0 respectively because of zero values in go.

Reply:
Right, I am going to not pass the empty string values to the tool (not an interesting scenario anyway)

Reply:
please see the proposed change

Reply:
Looks good, Thanks!